test/tools/cli/IonProcess.java

```diff
@@ -265,115 +266,111 @@ private static void processToEventFromIonStream(IonWriter ionWriterForOutput,
-                            ionWriterForOutput.setFieldName(fieldName);
-                        }
-                        if (annotations != null && annotations.length != 0) {
-                            ionWriterForOutput.setTypeAnnotationSymbols(annotations);
-                        }
-                        ionWriterForOutput.stepIn(type);
-                    }
-                } else if (event.getEventType().equals(EventType.CONTAINER_END)) {
-                    ionWriterForOutput.stepOut();
-                } else if (event.getEventType().equals(EventType.SCALAR)) {
-                    writeIonByType(event, ionWriterForOutput);
-                } else if (event.getEventType().equals(EventType.SYMBOL_TABLE)) {
-                    //TODO symbol table
-                    // For SYMBOL_TABLE events, flush the writer's existing local symbol table and any buffered data,
-                    // forcing the writer to create a new local symbol table that imports the list of symbol tables
-                    // declared by the imports field of the Event. This ensures that symbol tokens with unknown text
-                    // that occur in subsequent events in the stream can be written correctly
-                } else if (event.getEventType().equals(EventType.STREAM_END)) {
-                    ionWriterForOutput.finish();
+                                                       IonWriter ionWriterForErrorReport,
+                                                       IonReader ionReader,
+                                                       ProcessContext processContext,
+                                                       ProcessArgs args,
+                                                       OutputStream outputStream) throws IOException {
+        while (ionReader.next() != null) {
+            //update eventIndex
+            processContext.setEventIndex(processContext.getEventIndex() + 1);
+            Event event = eventStreamToEvent(ionReader);
+            //event.validate();
+            processContext.setLastEventType(event.getEventType());
+            if (event.getEventType() == EventType.CONTAINER_START) {
+
+                if (isEmbeddedEvent(event)) {
+                    embeddedEventToIon(ionReader, ionWriterForOutput, processContext, args, outputStream);
+                } else {
+                    IonType type = event.getIonType();
+                    setAnnotationAndField(event, ionWriterForOutput);
+                    ionWriterForOutput.stepIn(type);
                 }
+            } else if (event.getEventType().equals(EventType.CONTAINER_END)) {
+                ionWriterForOutput.stepOut();
+            } else if (event.getEventType().equals(EventType.SCALAR)) {
+                writeIonByType(event, ionWriterForOutput);
+            } else if (event.getEventType().equals(EventType.SYMBOL_TABLE)) {
+                ionWriterForOutput.finish();
```

Comment:
This should be `close()`, which finishes the writer and frees any resources it consumes.

Reply:
Changed

question: 

If I finish the writer and free outputStream of this writer, the outputStream will be closed so my new ionWriter can't write to the same outputStream anymore. Did I miss anything here?

Reply:
You can create a new OutputStream that appends to the existing file. See https://docs.oracle.com/javase/7/docs/api/java/io/FileOutputStream.html#FileOutputStream(java.io.File,%20boolean)